### ISSUE TITLE
`esc` escapes find before it escapes selection.

### ISSUE THREAD

**Issue by amano-kenji** [sxyazi/yazi#1752]
### What system are you running Yazi on?

Linux X11

### What terminal are you running Yazi in?

foot 1.16.2

### `yazi --debug` output

```Shell
not relevant
```

### Did you try the latest nightly build to see if the problem got fixed?

No, and I'll explain why below

### Describe the bug

I find a directory with `/`, enter it, and select a few files with space.

Then, I press `esc` to cancel selection. However, `find` is cancelled before selection is cancelled.

This behavior was puzzling to me.

### Minimal reproducer

Explained in the bug.

**Comment by amano-kenji** [comment]
This is relevant.

https://github.com/sxyazi/yazi/blob/558d97d08484d3d1279ac55ac7f1e3d845b5096c/yazi-core/src/tab/commands/escape.rs#L35

**Comment by sxyazi** [comment]
No, this is by design. If you first deselect and then cancel the find, it would mean that every time a user cancels find, they have to give up the list of files they've already selected.

Introducing a cancel stack doesn't make sense to me either because it would break the following workflow:

- Press `f` to search, then type a keyword like "a" to list all files containing "a".
- Press `Space` to select some files.
- Press `Esc` to exit the search mode and return to the original file list - but since your last action was selecting files, it actually does a deselect instead of going back to the file list.

If you want to change their order, please create a plugin to do that; here's an example https://github.com/sxyazi/yazi/pull/1042#issuecomment-2113994116. 

Also, if you just want to cancel the find when entering or leaving a directory, you can simply rebind `h` and `l`:

``` 
{ on = "h", run = ["leave", "escape --find"], desc = "Go back to the parent directory" },
{ on = "l", run = ["enter", "escape --find"], desc = "Enter the child directory" }, 
```

Closing as it's the expected behavior

**Comment by amano-kenji** [comment]
Okay.
